mecha.zig

```diff
@@ -493,12 +644,15 @@ pub fn convert(
     return .{ .parse = struct {
         fn parse(allocator: mem.Allocator, str: []const u8) Error!Res {
             const r = try parser.parse(allocator, str);
-            const v = conv(allocator, r.value) catch |e| switch (@as(anyerror, e)) {
-                error.ParserFailed => return error.ParserFailed,
-                error.OutOfMemory => return error.OutOfMemory,
-                else => return error.OtherError,
+            switch (r) {
+                .err => return Res.failWith(r, 0),
+                .ok => {},
+            }
```

Comment:
There's a lot of places where you switch on the result, but don't unpack the `ok` value. You then need to do `r.ok` later. Instead, do this:

```suggestion
            const r = switch (try parser.parse(allocator, str)) {
                .ok => |ok| ok,
                .err => |e| return e,
            };
```